tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/TracingIntegrationTest.cs

```diff
@@ -0,0 +1,25 @@
+// <copyright file="TracingIntegrationTest.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using Datadog.Trace.TestHelpers;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests
+{
+    public abstract class TracingIntegrationTest : TestHelper
+    {
+        public TracingIntegrationTest(string sampleAppName, ITestOutputHelper output)
+            : base(sampleAppName, output)
+        {
+        }
+
+        public TracingIntegrationTest(string sampleAppName, string samplePathOverrides, ITestOutputHelper output)
+            : base(sampleAppName, samplePathOverrides, output)
+        {
+        }
+
+        public abstract Result ValidateIntegrationSpan(MockSpan span);
```

Comment:
Anyway that this could be called automatically for each tests that inherits from this class? From a `Dispose` method maybe? I guess XUnit is not too helpful here ...

Reply:
Yeah, nice idea, but not sure that's doable with unfortunately - IIRC, the `Dispose` method runs _outside_ the context of a test, so failures there wouldn't be reported properly 🤔